src/main.rs

```diff
@@ -250,20 +252,60 @@ fn run_machete() -> anyhow::Result<bool> {
     Ok(has_unused_dependencies)
 }
 
-fn remove_dependencies(manifest: &str, dependencies_list: &[String]) -> anyhow::Result<String> {
+// handle a superset of all dependency name dashed/underscored variants: re'\w[-_]'
+fn dep_name_superset(dep_names: &[String]) -> HashSet<String> {
+    let mut unused: HashSet<String> = dep_names.iter().cloned().collect();
+    for dep in unused.clone() {
+        unused.insert(dep.replace('-', "_"));
+        unused.insert(dep.replace('_', "-"));
+    }
+    unused
+}
+
+fn remove_dependencies(manifest: &str, dependency_list: &[String]) -> anyhow::Result<String> {
     let mut manifest = toml_edit::Document::from_str(manifest)?;
-    let dependencies = manifest
+    let missing_table_msg = "unexpected missing table, please report with a test case on https://github.com/bnjbvr/cargo-machete";
+    let valid_tables = HashSet::from(["dependencies", "build-dependencies", "dev-dependencies"]);
+    let dependency_superset = dep_name_superset(dependency_list);
+
+    let mut dep_tables = manifest
         .iter_mut()
-        .find_map(|(k, v)| (v.is_table_like() && k == "dependencies").then_some(Some(v)))
-        .flatten()
-        .context("dependencies table is missing or empty")?
-        .as_table_mut()
-        .context("unexpected missing table, please report with a test case on https://github.com/bnjbvr/cargo-machete")?;
-
-    for k in dependencies_list {
-        dependencies
-            .remove(k)
-            .with_context(|| format!("Dependency {k} not found"))?;
+        // select for items that are tables with valid names
+        .filter(|(k, v)| v.is_table_like() && valid_tables.contains(k.display_repr().as_ref()))
+        .map(|(k, v)| {
+            let table = v.as_table_like_mut().context(missing_table_msg)?;
+            Ok((k, table))
+        })
+        .filter(|v| match v {
+            Ok((_, table)) if dependency_superset.iter().any(|k| table.contains_key(k)) => true,
+            Err(_) => true,
+            Ok(_) => false,
+        })
```

Comment:
```suggestion
Maybe `valid_tables` has some kind of `.get()` that would make it so we can avoid the filter, then the map, then the error handling step?

```suggestion
        .filter_map(|(k, v)| if v.is_table_like() {
          if let Some(table) = valid_tables.get(k.display_repr().as_ref())) {
            if dependency_superset.iter().any(|k| table.contains_key(k)) {
              Some((k, table))
            } else {
              None
            }
          } else {
            None
          }
        })
```

Or, even better, let's get rid of all the iterators transformers and use plain imperative code instead, as it might be more readable in this case.